File: weaveString.cpp
#include <iostream>
#include <string>
using namespace std;

string weaveString (string string1, string string2){
    int small = string1.length();
    string big_string = string2;
    string output_string = "";
    if (string2.length() < small){
        small = string2.length();
        big_string = string1;
    }
    for (int i = 0; i < small; i++){
        output_string = output_string + string1[i] + string2[i];
    }
    output_string = output_string + big_string.substr(small,big_string.length());
    return output_string;
}

int main(){
    cout<< weaveString("Yashi","Rocket");

    return 0;
}

File: assessment1/weaveString.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jun  7 18:02:32 2019

@author: yashi
"""

def weaveString (string1, string2):
    small = len(string1)
    big_string = string2
    output_string = ""
    if len(string2) < small:
        small = len(string2)
        big_string = string2
    for i in range(small):
        output_string = output_string + string1[i] + string2[i]
    output_string += big_string[small:] 
    return output_string

print(weaveString("Yashi","Rocket"))

